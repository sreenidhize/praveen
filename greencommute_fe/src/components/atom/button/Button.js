import React from "react";
import { Button as MButton, makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
  base: {
    borderRadius: "10px",
    fontWeight: "bold",
    fontSize: "14px",
    textTransform: "none",
    fontFamily: theme.typography.h5.fontFamily,
  },
  small: {
    width: "101px",
    height: "38px",
    padding: theme.spacing(1, 4),
  },
  medium: {
    width: "138px",
    height: "50px",
    padding: theme.spacing(2, 4),
  },
  xMedium: {
    width: "151px",
    height: "50px",
    padding: theme.spacing(1, 4),
  },
  long: {
    width: "208px",
    height: "50px",
    padding: theme.spacing(1, 4),
  },
  smallBorder: {
    borderRadius: "6px",
    width: "101px",
    height: "40px",
    padding: theme.spacing(1, 4),
  },
}));

const Button = ({
  variant,
  type,
  onClick,
  label,
  color,
  children,
  ...rest
}) => {
  const classes = useStyles();

  return (
    <MButton
      variant={variant}
      color={color}
      className={`${classes.base} ${classes[type]}`}
      onClick={onClick}
      {...rest}
    >
      {label || children}
    </MButton>
  );
};

Button.propTypes = {
  variant: PropTypes.oneOf(["contained", "outlined"]),
  type: PropTypes.string,
  color: PropTypes.string
};

Button.defaultProps = {
  variant: "contained",
  type: "medium",
  color: "primary",
};

export default Button;
