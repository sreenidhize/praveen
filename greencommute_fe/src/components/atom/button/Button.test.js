import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import Button from './Button'

describe("Button renders correctly", () => {
    test("should match snapshot", () => {
      const { asFragment } = render(<Button />);
      expect(asFragment()).toMatchSnapshot();
    });
  
    test("Button can be clicked", () => {
      let checked = false;
      const onClick = jest.fn();
  
      const { container } = render(
        <Button
          onClick={onClick}
          data-testid="button"
        />
      );
      screen.getByTestId("button");
      userEvent.click(container.querySelector("button"));
      expect(onClick).toHaveBeenCalled();
    });
  });
  