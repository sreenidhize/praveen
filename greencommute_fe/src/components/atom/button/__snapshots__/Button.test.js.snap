// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-base-1 makeStyles-medium-3 MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
